### PR TITLE
Siwa pparzer patch 1

### PR DESCRIPTION
**Pull request by siwa-pparzer**
respecting the storage page if constant `module.tx_newsletter.persistence.storagePid` is set

### CODE REVIEW

**Comment by PowerKiKi** [comment]
This PR does not respect coding style and thus fails on Travis. Please use the following to check and fix issues:
 ```
npm install -g gulp
npm install
gulp
```

**Comment by siwa-pparzer** [comment]
is this PR ok now?

**Comment by PowerKiKi** [comment]
I believe `hasStoragePid` should be `private` and a boolean, not an integer, and phpDocumented as such.

But most importantly the repository should be self-aware of its own configuration and not dependent on who is using it. So instead of "check settings in controller, configure repository and then use it", rather do "self-configure repository upon initialization". It seems that [this answer](http://typo3.3.n7.nabble.com/How-to-get-Plugin-settings-in-ExtBase-Repository-in-v6-2-td260265.html#a260280) should give us a clue on how to get settings from within the repository.

**Comment by siwa-pparzer** [comment]
the repository gets the storagePid now via objectManager
the controller is the same now as in the master branch
code checks seem good

File: Classes/Domain/Repository/BounceAccountRepository.php [hunk 1]
```diff
@@ -3,27 +3,27 @@
...
- *  http://www.gnu.org/copyleft/gpl.html.
- *
- *  This script is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  This copyright notice MUST APPEAR in all copies of the script!
- * ************************************************************* */
+     *  Copyright notice
```
Comment (PowerKiKi) [inline Classes/Domain/Repository/BounceAccountRepository.php]: This should not have been modified

**Comment by PowerKiKi** [comment]
Looks much better, thanks for the modifications. There is only that file header that should not have been indented.

The last thing we need is to document this new behavior in documentation. I guess in `Documentation/Configuration/Index.rst` would be fine.

And finally if you [squash all your commit into one](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request), then I'll merge.

**Comment by siwa-pparzer** [comment]
commits merged
all checks passed

**Comment by PowerKiKi** [comment]
Merged in 499403d768ccd3b2796b7f47a53adbda3fec3453, thanks for your work !

**Pull request closed without merging**